Remove CocoaPods

mokagio opened the pull request:
Picks up where #23750, but starting from scratch because of too many project file changes since then, and completes the migration away from CocoaPods as the mean of linking the Gutenberg XCFramework.

The sequence of changes is roughly:

- De-integrate CocoaPods
- Drag-n-drop the `xcframework`s
- Tweak build settings to work around compilation errors due to the frameworks see fbb0997a21089d01890222fa8568c9e67946c49c and 4a29aeacd770482bf998d01aa3bea74a09cfcf15
- Update the script that copies the React Native `App.js` file from the XCFramework to the build products as a `jsbundle` to use the new location
- Remove the various "if cocoapods" kind of checks from app code and automation

Two things noteworthy:

- I had to disable a couple of clang checks, `CLANG_WARN_STRICT_PROTOTYPES` and `CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER`, because of build failures in the XCFramework imports. It doesn't seem ideal, but also we have increasingly less Objective-C so it should be acceptable. Open to suggestions.
- @jkmassel 's implementation had a build phase that downloaded the XCFrameworks sources. I omitted it and **expect devs to run `make dependencies` before building the app**. It felt a build phase was a bit too much, but it also feels like a line in the `README` to call attention to this is not enough. Open to suggestions.

## Testing

I published this from the Simulator: https://giotestdotsite.wordpress.com/2025/01/07/via-gutenbegkit-without-cocoapods/   — Not sure if the HTML source can show it, but I did check locally that the experimental editor toggle was disabled. Besides, the UI being a bit different between the two helps being aware of it.

Apart from that, I'd say run the prototype build on device and see if the Gutenber-via-XCFramework editor works as expected.

## Regression Notes
1. Potential unintended areas of impact

It's possible that some behavior in the Gutenber-XCFramework editor will be compromised. However, I don't see why, once the editor loads, all the code should be the same.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Had a go at publishing a post from the Simulator. 

3. What automated tests I added (or what prevented me from doing so)

N.A.

---

PR submission checklist:

- [x] I have completed the Regression Notes. 
- [x] I have considered adding unit tests for my changes. - N.A.
- [x] I have considered adding accessibility improvements for my changes. - N.A.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. - N.A.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

mokagio committed:
Run `pod deintegrate` and remove CocoaPods automation

mokagio committed:
Add `Makefile` automation to download Gutenberg framework

mokagio committed:
Remove dead reference to `Pods.framework` in project

mokagio committed:
Remove dead reference to pods from workspace

mokagio committed:
Remove SwiftLint-via-CocoaPods from `Rakefile`

mokagio committed:
Remove unnecessary `$(inherited)` flag from a couple of targets

That's for the value `CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER`,
but it's useless to set `$(inherited)` for a Yes or No value because
removing the value from the target will automatically make Xcode read it
from the project, achieving the same result.

mokagio committed:
Remove CocoaPods instruction from the README

mokagio committed:
Remove SwiftLint instructions from the `README`

mokagio committed:
Add a version of Aztec fetched via SwiftPM

mokagio committed:
Add Gutenberg XCFrameworks via drag-n-drop from local paths

mokagio committed:
Disable `CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER`

Was giving errors in Yoga:

> Double-quoted include "YGEnums.h" in framework header, expected angle-bracketed instead

mokagio committed:
Disable `CLANG_WARN_STRICT_PROTOTYPES`

It resulted in a build failure:

> React.framework/Headers/RCTAppearance.h:16:60:
> A function declaration without a prototype is deprecated in all versions of C

mokagio committed:
Update search path for Gutenberg React Native bundle in script

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>Modules/Package.swift</c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Remove some leftover bits of CocoaPods set up for automation

mokagio committed:
Remove leftover `COCOAPODS` precompiler flag

Strange that `pod deintegrate` did not remove it. Maybe it was some
custom piece of setup...

mokagio committed:
Remove a couple of CocoaPods-specific code branches

mokagio committed:
Temporarily switch to CI toolkit branch compatible with no CocoaPods

mokagio committed:
Fetch XCFrameworks in CI

mokagio committed:
Update Gutenberg checkout logic in localization lint for CP-less setup

mokagio committed:
Update source map upload for CP-less Gutenberg setup

mokagio committed:
Ensure `Frameworks` folder exist, otherwise tar will fail

mokagio committed:
Remove "Log Outdated Pods" step from code freeze

We don't have pods anymore

mokagio committed:
Remove Pods from lists in `.gitignore` and RuboCop

We don't use Pods anymore

mokagio committed:
Remove imports of Pods xcconfigs

mokagio committed:
No longer look in Pods folder to extract localizations

mokagio commented on WordPress/Classes/System/WordPressAppDelegate.swift:
```diff
@@ -400,11 +400,6 @@ extension WordPressAppDelegate {
-        /// WordPressKit, now imported via CocoaPods, has the `AppExtension Safe API Only` flag set to *true*. Meaning that
-        /// the host app is, effectively as of now, responsible for presenting any alert onscreen (whenever a HTTP Challenge is
-        /// received). Capicci?
-        ///
```
This was a very old comment. It didn't seem to explain much to me, so I got rid of it.

mokagio committed:
Link Gutenberg XCFramework in Jetpack target too!

mokagio commented on WordPressAuthenticator/Sources/Private/WPAuthenticator-Swift.h:
```diff
@@ -1,8 +0,0 @@
-    #import "WordPressAuthenticator-Swift.h"
-#else
-    #import <WordPressAuthenticator/WordPressAuthenticator-Swift.h>
-#endif
```
This was only useful for the CocoaPods setup. I noticed it grepping for CocoaPods and removed it. There were only two usages anyway, which I updated to the generated header, see below.

mokagio committed:
Add instruction in README to run `make dependencies`

kean commented:
I hope it'll merge with https://github.com/wordpress-mobile/WordPress-iOS/pull/23923. There wasn't anyone available to review stuff, so that's why there is this trunk #2.

mokagio commented:
> I hope it'll merge with https://github.com/wordpress-mobile/WordPress-iOS/pull/23923. 

Love the `christmas-feature-branch` Nice work.

I'll try moving the changes there today.

mokagio commented:
@kean seems like everything worked fine. See https://github.com/wordpress-mobile/WordPress-iOS/pull/23958 — There are CI failures, but look unrelated to the changes, as they match what's in the base branch, example https://buildkite.com/automattic/wordpress-ios/builds/25312

mokagio commented:
> There are CI failures, but look unrelated to the changes, as they match what's in the base branch, example [buildkite.com/automattic/wordpress-ios/builds/25312](https://buildkite.com/automattic/wordpress-ios/builds/25312)

@kean I opened https://github.com/wordpress-mobile/WordPress-iOS/pull/23960 to start addressing them

mokagio commented:
Closing in favor of https://github.com/wordpress-mobile/WordPress-iOS/pull/23958
